packages/sdk/wallets/wallet-walletconnect/src/wc-wallet.ts

```diff
@@ -0,0 +1,147 @@
+import { sleep } from '@celo/base'
+import { CeloTx } from '@celo/connect'
+import { RemoteWallet } from '@celo/wallet-remote'
+import WalletConnect, { CLIENT_EVENTS } from '@walletconnect/client'
+import { ClientOptions, PairingTypes, SessionTypes } from '@walletconnect/types'
+import debugConfig from 'debug'
+import { SupportedMethods } from './types'
+import { WalletConnectSigner } from './wc-signer'
+
+const debug = debugConfig('kit:wallet:wallet-connect-wallet')
+
+/**
+ * Session establishment happens out of band so after somehow
+ * communicating the connection URI (often via QR code) we can
+ * continue with the setup process
+ */
+async function waitForTruthy(getValue: () => any, attempts: number = 10) {
+  let waitDuration = 500
+  for (let i = 0; i < attempts; i++) {
+    if (getValue()) {
+      return
+    }
+
+    await sleep(waitDuration)
+    waitDuration = waitDuration * 1.5
+  }
+
+  throw new Error('Unable to get pairing session, did you lose internet connection?')
+}
+
+/*
+ *   WARNING: This class should only be used with well-permissioned providers (ie IPC)
+ *   to avoid sensitive user 'privateKey' and 'passphrase' information being exposed
+ */
+export class WalletConnectWallet extends RemoteWallet<WalletConnectSigner> {
+  private options: ClientOptions
+  private metadata: SessionTypes.Metadata
+
+  private client?: WalletConnect
+  private pairing?: PairingTypes.Proposal
+  private session?: SessionTypes.Settled
+
+  constructor({ options, metadata }: { options?: ClientOptions; metadata: SessionTypes.Metadata }) {
+    super()
+
+    this.options = { relayProvider: 'wss://staging.walletconnect.org', ...options }
+    this.metadata = metadata
+  }
+
+  /**
+   * Get the URI needed for out of band session establishment
+   */
+  public async getUri() {
+    this.client = await WalletConnect.init(this.options)
+
+    this.client.on(CLIENT_EVENTS.session.proposal, this.onSessionProposal)
+    this.client.on(CLIENT_EVENTS.session.created, this.onSessionCreated)
+    this.client.on(CLIENT_EVENTS.session.updated, this.onSessionUpdated)
+    this.client.on(CLIENT_EVENTS.session.deleted, this.onSessionDeleted)
+
+    this.client.on(CLIENT_EVENTS.pairing.proposal, this.onPairingProposal)
+    this.client.on(CLIENT_EVENTS.pairing.created, this.onPairingCreated)
+    this.client.on(CLIENT_EVENTS.pairing.updated, this.onPairingUpdated)
+    this.client.on(CLIENT_EVENTS.pairing.deleted, this.onPairingDeleted)
+
+    this.client.connect({
+      metadata: this.metadata,
+      permissions: {
+        blockchain: {
+          // alfajores, mainnet, baklava
+          chains: ['celo:44787', 'celo:42220', 'celo:62320'],
```

Comment:
Are there cases where we'd want to connect to other chains?

Reply:
Yeah! Will allow passing those options through